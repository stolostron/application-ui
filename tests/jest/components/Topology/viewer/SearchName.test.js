/*******************************************************************************
 * Copyright (c) 2020 Red Hat, Inc.
 *******************************************************************************/
"use strict";

import Search from "../../../../../src-web/components/Topology/viewer/SearchName";
import renderer from "react-test-renderer";

const testSearchName = "test";
const locale = "en-US";

describe("SearchName search", () => {
  let btn = document.createElement("a");
  btn.innerHTML = "<button className='bx--search-close' />";
  btn.className = "bx--search-close";
  document.body.appendChild(btn);
  const props = {
    searchName: testSearchName,
    onNameSearch: jest.fn(),
    locale: locale
  };

  const searchInst = new Search(props);
  searchInst.setNameSearchRef(document);
  searchInst.handleSearch({ target: { value: "test" } });
  it("render as expected", () => {
    const component = renderer.create(searchInst.render());
    expect(component.toJSON()).toMatchSnapshot();
  });
});
