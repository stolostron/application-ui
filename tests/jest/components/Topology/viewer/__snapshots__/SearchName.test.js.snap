// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchName search render as expected 1`] = `
<div
  aria-label="Name"
  className="search-filter"
  id="Name"
  role="region"
>
  <div
    aria-labelledby="search-name-label"
    className="bx--search bx--search--lg"
    role="search"
  >
    <svg
      alt=""
      aria-label=""
      className="bx--search-magnifier"
      fillRule="evenodd"
      height="16"
      role="img"
      viewBox="0 0 16 16"
      width="16"
    >
      <title>
        
      </title>
      <path
        d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
      />
    </svg>
    <label
      className="bx--label"
      htmlFor="search-name"
      id="search-name-label"
    >
      
    </label>
    <input
      aria-label="Search-input"
      className="bx--search-input"
      id="search-name"
      onChange={[Function]}
      placeholder="Search [+Search]"
      type="text"
      value="test"
    />
    <button
      className="bx--search-close"
      onClick={[Function]}
      type="button"
    >
      <svg
        alt="Provide a description that will be used as the title"
        aria-label="Provide a description that will be used as the title"
        fillRule="evenodd"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
      >
        <title>
          Provide a description that will be used as the title
        </title>
        <path
          d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414 8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
        />
      </svg>
    </button>
  </div>
</div>
`;
